src/systems/detachable_joint/DetachableJoint.cc

```diff
@@ -234,8 +234,33 @@ void DetachableJoint::PreUpdate(
     }
     else
     {
-      modelEntity = _ecm.EntityByComponents(
-          components::Model(), components::Name(this->childModelName));
+      auto candidateEntities = entitiesFromScopedName(
+        this->childModelName, _ecm);
+
+      if (candidateEntities.size() == 1)
+      {
+        // If there is one entity select that entity itself
+        modelEntity = *candidateEntities.begin();
+      }
+      else
+      {
+        auto parentEntityScopedPath = scopedName(this->model.Entity(), _ecm);
+        // If there is more than one entity with the same name, look for the
+        // entity with a parent component.
+        for (auto entity : candidateEntities)
+        {
+          auto childEntityScope = scopedName(entity, _ecm);
+          if (childEntityScope.size() < parentEntityScopedPath.size())
+          {
+            continue;
+          }
+          if (childEntityScope.find(parentEntityScopedPath) != 0)
+          {
+            continue;
+          }
+          modelEntity = entity;
```

Comment:
If there are multiple possible child models even after this name resolution, I'd suggest we log a warning to the user and mention the scoped path of the selected child model.

Reply:
[e8e1b16](https://github.com/gazebosim/gz-sim/pull/1097/commits/e8e1b16976b2774f3f9cbade1eb8c0978aa3279a)